src/renderers/canvas/sigma.canvas.labels.def.js

```diff
@@ -86,10 +86,68 @@
         break;
     }
 
-    context.fillText(
-      node.label,
-      Math.round(node[prefix + 'x'] + labelOffsetX),
-      Math.round(node[prefix + 'y'] + labelOffsetY)
-    );
+    var lines = getLines(node.label, maxLineLength),
+        baseX = node[prefix + 'x'] + labelOffsetX,
+        baseY = Math.round(node[prefix + 'y'] + labelOffsetY);
+
+    for (var i = 0; i < lines.length; ++i) {
+      context.fillText(lines[i], baseX, baseY + i * (fontSize + 1));
+    }
   };
+
+  function getLines(text, maxLineLength) {
+    if (maxLineLength <= 1) {
+      return [text];
+    }
+
+    var words = text.split(' '),
+        lines = [],
+        lineLength = 0,
+        lineIndex = -1,
+        lineList = [],
+        lineFull = true;
+
+    for (var i = 0; i < words.length; ++i) {
+      if (lineFull) {
+        if (words[i].length > maxLineLength) {
+          var parts = splitWord(words[i], maxLineLength);
+          for (var j = 0; j < parts.length; ++j) {
+            lines.push([parts[j]]);
+            ++lineIndex;
+          }
+          lineLength = parts[parts.length - 1].length;
+        } else {
+          lines.push([words[i] + ' ']);
+          ++lineIndex;
+          lineLength = words[i].length + 1;
+        }
+        lineFull = false;
+      } else if (lineLength + words[i].length <= maxLineLength) {
+        lines[lineIndex].push(words[i] + ' ');
+        lineLength += words[i].length + 1;
+      } else {
+        lineFull = true;
+        --i;
+      }
+    }
+
+    for (i = 0; i < lines.length; ++i) {
+      lineList.push(lines[i].join(''))
```

Comment:
optimization: use `join(' ')` and use `words[i]` instead of `words[i] + ' '` above